portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -136,10 +215,41 @@ private PreparedQuery getAllComments() {
     return datastore.prepare(query);
   }
 
+  /** 
+    * Returns all user comments that contain `search` in either
+    * the comment text or author name. 
+    * @param results All available comment entities in the database.
+    * @param comments An empty list to be filled. 
```

Comment:
Instead of requiring all method callers to give an empty list, you could simply generate an empty array list inside the method.

Reply:
See reply below :)